posthog/session_recordings/session_recording_api.py

```diff
@@ -292,6 +295,56 @@ def list(self, request: request.Request, *args: Any, **kwargs: Any) -> Response:
         filter = SessionRecordingsFilter(request=request, team=self.team)
         return list_recordings_response(filter, request, self.get_serializer_context())
 
+    @extend_schema(
+        exclude=True,
+        description="""
+        Gets the comments associated with a single recording from any notebooks""",
+    )
+    @action(methods=["GET"], detail=True)
+    def comments(self, request: request.Request, *args: Any, **kwargs: Any) -> JsonResponse:
+        recording: SessionRecording = self.get_object()
+
+        with connection.cursor() as cursor:
+            query = """
+                SELECT
+                    short_id,
+                    title,
+                    string_agg(text_elem->>'text', ' ') AS comment,
+                    timestamp_elem->'attrs'->>'playbackTime' AS timeInRecording
+                FROM
+                    posthog_notebook,
+                    LATERAL jsonb_array_elements(content->'content') AS timestamp_json,  -- Explode the top-level content array
+                    LATERAL jsonb_array_elements(timestamp_json->'content') AS timestamp_elem -- Explore deeper content
+                    LEFT JOIN LATERAL jsonb_array_elements(timestamp_json->'content') AS text_elem ON text_elem->>'type' = 'text'
+                WHERE
+                    timestamp_elem->>'type' = 'ph-replay-timestamp'
+                    AND team_id = %s
+                    AND jsonb_extract_path_text(timestamp_elem->'attrs', 'sessionRecordingId') = %s
+                GROUP BY team_id, title, short_id, timeInRecording;
+            """
+
+            cursor.execute(query, [self.team_id, recording.session_id])
+            results = cursor.fetchall()
+            recording_comments = []
+            for result in results:
+                short_id = result[0]
+                title = result[1]
+                comment = result[2]
+                time_in_recording = result[3]
+                # the individual comments don't have an id, so we'll generate one
```

Comment:
What is the id needed for?

Reply:
iirc everything in the inspector has an ID so that e.g. expanded state is correct on re-render
added it here instead of a uuid or something artificially in the UI